homeassistant/components/met_eireann/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""The met_eireann component."""
+from datetime import timedelta
+import logging
+from random import randrange
+
+import meteireann
+
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    LENGTH_FEET,
+    LENGTH_METERS,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util.distance import convert as convert_distance
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured Met Éireann."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up Met Éireann as config entry."""
+    coordinator = MetEireannDataUpdateCoordinator(hass, config_entry)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(config_entry, "weather")
+    )
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+    await hass.config_entries.async_forward_entry_unload(config_entry, "weather")
+    hass.data[DOMAIN].pop(config_entry.entry_id)
+
+    return True
+
+
+class MetEireannDataUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
Why do we need a separate class here instead of just instantiating a DataUpdateCoordinator object directly?  It doesn't look like any additional functionality is added here.